src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.component.ts

```diff
@@ -421,6 +451,20 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, OnDestroy
         this.showExistingQuestions = !this.showExistingQuestions;
         this.selectedCourseId = null;
         this.allExistingQuestions = this.existingQuestions = [];
+        this.cacheValidation();
+    }
+
+    /**
+     * @function cacheValidation
+     * @desc Check the validity of the quiz on change
```

Comment:
Please provide more detailed comments what is happening here. This is way too short!